### PR TITLE
Correctly handle empty environment files

### PR DESCRIPTION
**Pull request by psss**
If the environment file does not contain any variables we should
not traceback but handle this as an empty environment. Fix this
for both `--environment @file` and `--environment-file` options.

### CODE REVIEW

**Review by lukaszachy — APPROVED** [review]

**Comment by lukaszachy** [comment]
I guess we shouldn't merge this until we fix TF and read this file just once - otherwise steps could run with wrong environment which might get unnoticed.

**Comment by psss** [comment]
> I guess we shouldn't merge this until we fix TF and read this file just once - otherwise steps could run with wrong environment which might get unnoticed.

Yeah, good point!

**Review by thrix — APPROVED** [review]
LGTM

**Comment by psss** [comment]
Hmmm, thinking about it a bit more: Would it make more sense to always raise an error when the environment file is empty? Looking at it from the other side: User wanted to load environment from a file, so we can expect that the file should contain at least one variable definition. Why would the user otherwise provide the option? @lukaszachy, @thrix, thoughts?

**Comment by kkaarreell** [comment]
Not an error but warning might be appropriate.

**Comment by psss** [comment]
> Not an error but warning might be appropriate.

Makes sense, will update the pull request.

**Comment by psss** [comment]
The warning has been added and tests adjusted accordingly. In the second commit dfd9e45 I've improved gathering of environment variables from the command line so that the environment files are always read only once. This should unblock merging this pull request as during the run no other environment files access is done any more and if there is a problem with reading environment variables it will be clear from the warning and from the debug log where plan `environment` is logged.

@lukaszachy, @thrix, please have a look. Thanks.

**Comment by lukaszachy** [comment]
LGTM

**Pull request merged**
